screens/PostList.js

```diff
@@ -0,0 +1,45 @@
+import { firebase } from "../firebase";
+import { useState, useEffect } from "react";
+import { StyleSheet, ScrollView, TouchableOpacity } from "react-native";
+import { ListItem } from "react-native-elements";
+import { getDatabase, ref, onValue } from "firebase/database";
+import descendDateSort from "../helper/descendDateSort";
+import { useDispatch } from "react-redux";
+import { storePosts } from "../store/actions/posts";
+
+const PostList = ({ navigation }) => {
+  const dispatch = useDispatch();
+  const [postsData, setPostsData] = useState();
+
+  useEffect(() => {
+    const db = getDatabase();
+    const reference = ref(db, "/posts");
+    onValue(reference, (snapshot) => {
+      const data = snapshot.val();
+      setPostsData(descendDateSort(data));
+      dispatch(storePosts(data));
+    });
+  }, []);
+
+  return (
+    <ScrollView style={{ width: "100%" }}>
+      {postsData?.map((post, i) => (
+        <TouchableOpacity
+          key={`posts-${i}`}
```

Comment:
we shouldn't use index as a key

Reply:
for react rendering performance, we should use id for key, not index number in loop